src/server/handlers/launchpad.js

```diff
@@ -437,6 +439,21 @@ export const findSnap = (req, res) => {
     .catch((error) => sendError(res, error));
 };
 
+const clearSnapCache = (repositoryUrl) => {
+  const repository = parseGitHubUrl(repositoryUrl);
+  const enabledReposCacheId = getUrlPrefixCacheId(getRepoUrlPrefix(repository.owner));
+  const snapCacheId = getRepositoryUrlCacheId(repositoryUrl);
+  const snapNameCacheId = getSnapNameCacheId(repositoryUrl);
+
+  logger.info(`Clearing caches for ${repositoryUrl}: ${enabledReposCacheId}, ${snapCacheId}, ${snapNameCacheId}`);
+
+  return Promise.all([
+    getMemcached().del(enabledReposCacheId),
+    getMemcached().del(snapCacheId),
+    getMemcached().del(snapNameCacheId)
+  ]);
+};
```

Comment:
These are caches for 2 different LP API calls. One is listing of all snaps (enabled repos) for given user, and one is a single snap for given repository.

Yes, this data overlaps (for one user same snap is kept in both), but optimizing it may introduce unnecessary complexity in cache layer.